crates/data_components/src/duckdb.rs

```diff
@@ -288,19 +325,34 @@ impl DuckDB {
         result
     }
 
-    fn insert_table_into(&self, tx: &Transaction<'_>, table_to_insert_into: &DuckDB) -> Result<()> {
-        let insert_sql = format!(
+    fn insert_table_into(
+        &self,
+        tx: &Transaction<'_>,
+        table_to_insert_into: &DuckDB,
+        on_conflict: Option<&OnConflict>,
+    ) -> Result<()> {
+        let mut insert_sql = format!(
             r#"INSERT INTO "{}" SELECT * FROM "{}""#,
             table_to_insert_into.table_name, self.table_name
         );
 
+        if let Some(on_conflict) = on_conflict {
```

Comment:
This is where we finally consume the `OnConflict` struct if it exists and modify the INSERT SQL to have the proper on conflict clause.

Reply:
Note this function is only called by DuckDB when inserting into a table with constraints, otherwise the `insert_batch_no_constraints` method below is called.